openwisp_monitoring/monitoring/models.py

```diff
@@ -569,13 +569,15 @@ class Threshold(TimeStampedEditableModel):
     _THRESHOLD_OPERATORS = (('<', _('less than')), ('>', _('greater than')))
     metric = models.OneToOneField(Metric, on_delete=models.CASCADE)
     operator = models.CharField(max_length=1, choices=_THRESHOLD_OPERATORS)
-    value = models.IntegerField(help_text=_('threshold value'))
+    value = models.FloatField(help_text=_('threshold value'))
     seconds = models.PositiveIntegerField(
         default=0, validators=[MaxValueValidator(604800)], help_text=_(_SECONDS_HELP)
     )
 
     def _value_crossed(self, current_value):
         threshold_value = self.value
+        if type(current_value) == int:
+            current_value = float(current_value)
```

Comment:
is this necessary?

```
>>> 1.0 > 1
False
>>> 1.1 > 1
True
>>> 1 > 1.0
False
>>> 1 == 1.0
True
```

Also, if the data is converted to float there's no need to do this.

Reply:
I was thinking likewise, shall I then change this 
```python
method = '__gt__' if self.operator == '>' else '__lt__'
        return getattr(current_value, method)(threshold_value)

```

Reply:
does it work without this addition? If it does, I don't think is necessary. Am I missing something?
Where did the need for this change arise?

Reply:
@nemesisdesign, I added to take care of this,
```python
>>> getattr(1, '__gt__')(1.00)
NotImplemented
```
We want a `bool` value, if `NotImplemented` is returned it would raise a `ValueError`

Reply:
I didn't know that!